### PR TITLE
[IT-2407] redirect http to https

### PR DESCRIPTION
**Pull request by zaro0508**
Setup the http to https redirect in node[1]

[1] https://www.geeksforgeeks.org/how-to-detect-http-or-https-then-force-redirect-to-https-in-javascript/

### CODE REVIEW

File: src/server/server.ts [hunk 1]
```diff
@@ -30,11 +30,9 @@ app.set('trust proxy', 'loopback');
 
 //http to https redirects
 app.enable('trust proxy');
-app.use((req,res,next)=>{
-  if(rec.secure){
-    next();
-  }else{
-    res.redirect('https://'+req.host+req.url);
+app.use(function(request, response, next)=>{
+  if (request.protcol == 'http') {
```
Comment (xschildw) [inline src/server/server.ts]: protocol

Q: can't we just add listener to the ALB?

Comment (zaro0508) [reply src/server/server.ts]: yes, however AWS does not make it easy to automate and that's why i have decided to try to do it in node.

**Review by xschildw — APPROVED** [review]

**Comment by JessterB** [comment]
Unfortunately I don't think that this change will address our root issue, which is that the 'Share URL' button does not work when this page is loaded over http: http://agora.adknowledgeportal.org/genes/comparison

It fails because we can't access `navigator.clipboard` over http: https://stackoverflow.com/questions/51805395/navigator-clipboard-is-undefined

I believe that this PR would only redirect requests to the server (/api/\*), however we need the web app's URLs (/* pages) to be https to access `navigator.clipboard` on the page.

Note that the app already makes https requests to the server, even when the app itself is running over http:

Screen Shot 2023-01-11 at 11 03 53 AM

The stackoverflow thread I linked above also includes a workaround using an off-screen text area, but that solution requires using a deprecated command (https://developer.mozilla.org/en-US/docs/Web/API/Document/execCommand) that has been replaced by the `navigator.clipboard` API...

**Comment by JessterB** [comment]
This might be worth doing anyway if we want to prevent users from hitting our server API directly over http, although I don't know if anyone does that, or if we care about insecure access to our read-only APIs that return only open public data.

**Review by JessterB — APPROVED** [review]
Let's see what this does in dev, since validating it on localhost would require a fair amount of additional effort.

**Pull request merged**